### PR TITLE
Same Token pair with multiple fees implementation 

### PR DESCRIPTION
**Pull request by abhishek-k-sharma-lab49**

### CODE REVIEW

File: deployment/state/contracts.json [hunk 1]
```diff
@@ -257,5 +257,14 @@
         "hash": "6da23e6aa07b521fb8630769ebe02d8e",
         "transparentProxyAddress": "0x000000000000000000000000000000000032cb1d",
         "transparentProxyId": "0.0.3328797"
+    },
```
Comment (shivamgarglab49) [inline deployment/state/contracts.json:260]: please replace factory copy in UAT.json too because of new proxy.

File: test/pair-test.ts [hunk 2]
```diff
@@ -228,19 +229,32 @@ describe("All Tests", function () {
   });
 
   describe("Factory Contract positive Tests", async () => {
-    it("Check createPair method", async function () {
+    it("Check createPair method, Same Tokens and same fees", async function () {
```
Comment (amandeeprehal-lab49) [inline test/pair-test.ts:232]: Pls make it more descriptive what are you trying to test. 

Use Given, When and Then form

File: test/pair-test.ts [hunk 3]
```diff
@@ -1204,6 +1218,7 @@ describe("All Tests", function () {
       expect(value[0]).to.be.equals(token1Address);
       expect(value[1]).to.be.equals(token2Address);
       expect(value[2]).to.not.be.equals(newZeroAddress);
+      expect(value[3]).to.be.equals(fee);
     });
```
Comment (amandeeprehal-lab49) [inline test/pair-test.ts:1222]: Do we test case where tokenA, tokenB and fee are same i.e. duplicate check if not pls add?

File: integrationTest/factory.ts [hunk 4]
```diff
@@ -82,18 +83,22 @@ const swapToken = async (token: TokenId) => {
 };
 
 async function main() {
-  await factory.setupFactory();
-  await testForSinglePair(tokenB, tokenHBARX, new BigNumber(10));
-  await testForSinglePair(tokenB, tokenHBARX, new BigNumber(15));
-  await testForSinglePair(tokenB, tokenC);
+  try {
+    await factory.setupFactory();
```
Comment (shivamgarglab49) [inline integrationTest/factory.ts:87]: no try catch needed.

setupFactory method itself having try/catch.

File: e2e-test/business/Pair.ts [hunk 5]
```diff
@@ -200,10 +200,12 @@ export default class Pair extends Base {
     const tokenAAddress = result.getAddress(0);
     const tokenBAddress = result.getAddress(1);
     const lpTokenAddress = result.getAddress(2);
+    const fee = result.getInt256(3);
     console.log(`- Pair#${GET_TOKEN_PAIR_ADDRESS}():`);
     console.log(` - A Token Address = ${tokenAAddress}`);
     console.log(` - B Token Address = ${tokenBAddress}`);
     console.log(` - Lp Token Address = ${lpTokenAddress}\n`);
+    console.log(` - Fee Value = ${fee}\n`);
```
Comment (shivamgarglab49) [inline e2e-test/business/Pair.ts:208]: please return fee value as well.

caller can use it if needed.

**Review by shivamgarglab49 — APPROVED** [review]

File: contracts/mock/MockBaseContract.sol [hunk 6]
```diff
@@ -8,22 +8,33 @@ import "./ERC20Mock.sol";
 import "hardhat/console.sol";
 
 contract MockBaseHTS is IBaseHTS {
-    bool internal tokenTest;
-    bool private revertCreateToken;
-    int256 private passTransactionCount = 100;
     address private hbarx;
+    bytes32 revertCreateTokenSlot = keccak256("revertCreateTokenSlot");
```
Comment (shivamgarglab49) [inline contracts/mock/MockBaseContract.sol:12]: we can make them constants.

File: integrationTest/factory.ts (hunk 4 above)
Comment (amandeeprehal-lab49) [reply integrationTest/factory.ts]: Done

File: contracts/mock/MockBaseContract.sol (hunk 6 above)
Comment (amandeeprehal-lab49) [reply contracts/mock/MockBaseContract.sol:12]: It's local for this Mock contract and used only in one place so constant might not be useful.

File: e2e-test/business/Pair.ts (hunk 5 above)
Comment (amandeeprehal-lab49) [reply e2e-test/business/Pair.ts:208]: done

File: deployment/state/contracts.json (hunk 1 above)
Comment (amandeeprehal-lab49) [reply deployment/state/contracts.json]: Done

File: test/pair-test.ts (hunk 3 above)
Comment (amandeeprehal-lab49) [reply test/pair-test.ts:1338]: Done

**Review by amandeeprehal-lab49 — APPROVED** [review]

**Pull request merged**